Src/HALALMock/Services/Communication/SPI/SPI.cpp

```diff
@@ -62,6 +63,37 @@ uint8_t SPI::inscribe(SPI::Peripheral& spi) {
     SPI::registered_spi[id] = spi_instance;
     SPI::registered_spi_by_handler[spi_instance->hspi] = spi_instance;
 
+    // Create socket for this SPI peripheral
+    int spi_socket = socket(AF_INET, SOCK_STREAM, 0);
+
+    // Bind this socket to the local port 200x
+    sockaddr_in local_address;
+    local_address.sin_family = AF_INET;
+    local_address.sin_port = htons(SPI_PORT_BASE + id_counter);
+    local_address.sin_addr.s_addr = INADDR_ANY;
+
+    if (bind(spi_socket, (struct sockaddr*)&local_address,
+             sizeof(local_address)) < 0) {
+        ErrorHandler("Bind error: %s", strerror(errno));
+        close(spi_socket);
+        return 0;
+    }
+
+    // Connect this peripheral with slave simulator
+    sockaddr_in server_address;
+    server_address.sin_family = AF_INET;
+    server_address.sin_port = htons(2000 + id_counter);
+    server_address.sin_addr.s_addr = INADDR_ANY;
+
+    if (connect(spi_socket, (struct sockaddr*)&server_address,
+                sizeof(server_address)) < 0) {
+        ErrorHandler("Connect error: %s", strerror(errno));
+        close(spi_socket);
+        return 0;
+    }
```

Comment:
this should be a server, I believe connect blocks until the connection is made, so if the user just wants one SPI instance but the code declares 6, the first will connect but the other five ones will wait for a connection that won't be made

Reply:
But a socket only creates and tries to connect when a SPI peripheral REGISTERS, that means when the user is going to use it. Not the six sockets is going to create unless the user want to use them all

Reply:
I believe a possible solution is: if the peripheral is MASTER_MODE, launch a server, otherwise just become a client.

and I think @jorgesg82 is rith, you only open the socket when calling spi::inscribe, so that should be fine

Reply:
I recoded this part.

Now a peripheral that acts as a master creates a server socket, while a peripheral that acts as a salve creates a client socket.

Master peripherals use a pin as a Slave Select, which turns on when the peripheral start. I used this to tell the simulator when to connect, putting a bool true in the emulated pin.

The ST-LIB is coded to connect only one slave to one master, so when the peripheral starts the server only waits to ONE slave to connect

Reply:
I don't see the logic you mention, the SPI master should launch a server and wait for outside connections, maybe from another sensor/ from the Python generator, or another board (LCU-BCU).

If it's a slave it should block until successfully connecting to a master

Reply:
there is no `listen()` call?